admin/controllers/school.js

```diff
@@ -237,30 +234,27 @@ const postSubmitAttendance = async (req, res, next) => {
   if (!attendees) {
     return res.redirect('/school/submit-attendance')
   }
-  const data = Object.values(req.body[ 'attendee' ] || null)
-  let selected
-  const { pupils } = await pupilDataService.getPupils(req.user.School)
-  try {
-    let ids = data.map(id => mongoose.Types.ObjectId(id))
-    selected = await Pupil.find({ _id: { $in: ids } }).exec()
-  } catch (error) {
-    return next(error)
-  }
+  const data = Object.values(req.body[ 'attendee' ] || [])
+  let ids = data.map(id => mongoose.Types.ObjectId(id))
+
   // Update attendance for selected pupils
-  selected.forEach((p) => (p.hasAttended = true))
-  const selectedPromises = selected.map(s => s.save())
+  await pupilDataService.update2(
```

Comment:
Why don't we just create a new pupilDataservice to keep the new methods separated?

Reply:
Actually I didn't mean to leave that in there. :(  Good catch 👍